index.ts

```diff
@@ -35,10 +41,11 @@ class Client {
     }
   }
 
-  static async createChannel({ fetch = global.fetch } = {}) {
+  static async createChannel({ fetch = undiciFetch } = {}) {
     const response = await fetch("https://smee.io/new", {
       method: "HEAD",
       redirect: "manual",
+      dispatcher: new EnvHttpProxyAgent(),
```

Comment:
we could instantiate the proxy agent only once and reuse it also for the eventsource

Reply:
Done
13aad6257a39b0e27a7bbae3bc14f0313e655962